Ullaakut/Bloggo
# #1 Add base structure of the Bloggo

## Goal of the PR

Set the base structure of the project, with a few controllers, services & repos to organize the different parts of the API. Most of the logic is commented out at the moment and replaced by the most basic logic possible to make it possible to test already.

Fixes #1

<img width="953" alt="screenshot 2018-08-01 at 22 42 23" src="https://user-images.githubusercontent.com/6976628/43547759-7f577584-95dc-11e8-9312-2544ba95906f.png">

Currently instead of parsing the token, for the sake of simplicity, it just directly gets the `sub` claim from it through the header and checks whether or not said subject is registered (currently hard coded instead of in a database, since not sure what database to use yet)

## How to test it?

Run `dep ensure`, followed by `go run *.go` in the root of the repository. Then, in another terminal, run:

```
curl -X POST \
        http://0.0.0.0:4242/posts \
        -H 'Authorization: Bearer auth0|596f27c2c3709661e9cea37d' \
        -H 'Cache-Control: no-cache' \
        -H 'Postman-Token: ddf103dc-3ba5-4f08-adc3-6b9625b7581b'
```

This should return a 200 OK and an empty body.

Then try doing the same but replacing the token id with something invalid, and make sure that you get an error message and a `401 Unauthorized` response.
